packages/js/src/ui/helpers/useUncontrolledState.ts

```diff
@@ -0,0 +1,33 @@
+import { Accessor, createSignal } from 'solid-js';
+
+type UseUncontrolledState = {
+  value?: boolean;
+  fallbackValue?: boolean;
+};
+
+type UseUncontrolledStateOutput = [Accessor<boolean>, (value: boolean) => void];
+
+export function useUncontrolledState({
+  value,
+
+  fallbackValue,
+}: UseUncontrolledState): UseUncontrolledStateOutput {
+  const [uncontrolledValue, setUncontrolledValue] = createSignal(!!fallbackValue);
+
+  const handleUncontrolledChange = (val: boolean) => {
+    setUncontrolledValue(val);
+  };
+  /**
+   * If value is provided, return controlled state
+   */
+  if (value !== undefined) {
+    const accessor: Accessor<boolean> = () => value;
```

Comment:
I think you can just do a memo here to get the type. Otherwise, no reason to type this as `Accessor`.

Reply:
CreateMemo also returns an Accessor.
Accessor Type is just a simple Generic
`type Accessor<T> = () => T`
same as writing `() => boolean`
I can update the type to 
`() => boolean` if we decide to use that

Reply:
I was just saying, there's no need to type it out here. you can just do
`const accessor = () => value;` or `createMemo()`